src/plugins/tlsmon/private.h

```diff
@@ -0,0 +1,181 @@
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef TLSMON_PRIVATE_H
+#define TLSMON_PRIVATE_H
+
+#include <sstream>
+#include <iomanip>
+
+
+namespace tlsmon_priv
+{
+
+constexpr uint32_t NCRYPT_SSL_KEY_MAGIC_BYTES = 0x44444442;
+constexpr uint32_t MASTER_SECRET_MAGIC_BYTES  = 0x73736c35;
+
+constexpr size_t CLIENT_RANDOM_SZ = 0x20;
+constexpr size_t MASTER_KEY_SZ    = 0x30;
+
+/* Could not find whole structure documentation. We are only interested in
+magic bytes and pointer to master_secret structure. */
+typedef struct
+{
+    char field0[4];
+    uint32_t magic;
+    char field2[8];
+    void* master_secret;
+} __ncrypt_ssl_key_t;
+
+typedef struct
+{
+    uint32_t cb_struct_length;
+    uint32_t magic;
+    uint32_t protocol_version;
+    char aligment[4];
+    void* cipher_suite_list_entry;
+    uint32_t is_client_cache;
+    char master_key[MASTER_KEY_SZ];
+    uint32_t field7;
+} __ssl_master_secret_t;
+
+enum __ncrypt_buffer_type_t
```

Comment:
Please do not name stuff starting with `_` as those names are reserved for compiler.